robomaker-sample-app-ci/src/aws-robomaker-sample-application-ci.ts

```diff
@@ -65,127 +65,109 @@ function getExecOptions(workingDir, extraPath, listenerBuffers?): ExecOptions {
-        break;
-      }
-      console.log(`Colcon bundle failed.. retrying in ${delay_ms} milliseconds`);
-      await delay(delay_ms); // wait for next retry per the current exponential backoff delay
-      delay_ms = Math.min(delay_ms * 2,  1000 * MAXIMUM_BACKOFF_TIME_SECONDS); // double the delay for the next retry, truncate if required
-    }
-  }
+  // indexed from 0 because RETRIES is the number of retries AFTER the initial try
+  const bundleFilename = path.basename(WORKSPACE_DIRECTORY);
+  await exec.exec("colcon", ["bundle", "--build-base", "build", "--install-base", "install", "--bundle-base", "bundle"], getWorkingDirExecOptions());
+  await exec.exec("mv", ["bundle/output.tar", `../${bundleFilename}.tar`], getWorkingDirExecOptions());
+  await exec.exec("rm", ["-rf", "bundle"], getWorkingDirExecOptions());  // github actions have been failing with no disk space
 }
 
 async function run() {
+  let delay_ms = 1000 * MINIMUM_BACKOFF_TIME_SECONDS;
   console.log(`ROS_DISTRO: ${ROS_DISTRO}`);
   console.log(`WORKSPACE_DIRECTORY: ${WORKSPACE_DIRECTORY}`);
   console.log(`GENERATE_SOURCES: ${GENERATE_SOURCES}`);
-  console.log(`COLCON_BUNDLE_RETRIES: ${COLCON_BUNDLE_RETRIES}`);
+  console.log(`RETRIES: ${RETRIES}`);
   
-  // check if COLCON_BUNDLE_RETRIES is valid (i.e. 0<) and not too large (i.e. <10)  
-  if (COLCON_BUNDLE_RETRIES<0 || 9<COLCON_BUNDLE_RETRIES){
-    core.setFailed(`Invalid number of colcon bundle retries. Must be between 0-9 inclusive`);
+  // check if RETRIES is valid (i.e. 0<) and not too large (i.e. <10)  
+  if (RETRIES<0 || 9<RETRIES){
+    core.setFailed(`Invalid number of retries. Must be between 0-9 inclusive`);
   }
 
-  await setup();
-  if (GENERATE_SOURCES == 'true') {
-    await prepare_sources();
+  for (let i = 0; i <= RETRIES; i++) {
+    try {
+      await setup();
+      if (GENERATE_SOURCES == 'true') {
+        await prepare_sources();
+      }
+      await build();
+      await bundle();
+      core.setOutput('ros-distro', ROS_DISTRO)
+      core.setOutput('sample-app-version', SAMPLE_APP_VERSION);
+      break; //Break out of retry loop if successful.
+    } catch (error) {
+      await exec.exec("rm", ["-rf", "bundle"], getWorkingDirExecOptions()); // remove erred bundle assets
+      if (i == RETRIES){
+        core.setFailed(error.message); // set action to Failed after RETRIES number of retries
+        break;
+      }
+      console.log(`Action failed.. retrying in ${delay_ms} milliseconds`);
+      await delay(delay_ms); // wait for next retry per the current exponential backoff delay
+      delay_ms = Math.min(delay_ms * 2,  1000 * MAXIMUM_BACKOFF_TIME_SECONDS); // double the delay for the next retry, truncate if required
+      core.setFailed(error.message);
+    }
   }
-  await build();
-  await bundle();
 
-  core.setOutput('ros-distro', ROS_DISTRO)
-  core.setOutput('sample-app-version', SAMPLE_APP_VERSION);
```

Comment:
this output field is being used downstream in multiple places -- [here](https://github.com/aws-robotics/aws-robomaker-sample-application-helloworld/blob/ros1/.github/workflows/ros1.yml#L62) for instance